### PR TITLE
etcd_docker 4: Incorporate docker based etcd into Go integration tests

### PR DESCRIPTION
**Pull request by andrewmains12**
PR 4 for https://github.com/m3db/m3/issues/4144

High level approach is as described in https://github.com/m3db/m3/issues/4144 .

This PR integrates docker based etcd into our Go integration tests. It removes the need to have the embed package running
in m3db for them, but doesn't yet touch that functionality.

commit-id:3ae12ffd

---

**Stack**:
- #4150
- #4149
- #4148 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: src/integration/resources/docker/config/prometheus.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  scrape_interval:     5s # Set the scrape interval to every 5 seconds. Default is every 1 minute.
+  evaluation_interval: 5s # Evaluate rules every 5 seconds. The default is every 1 minute.
+  # scrape_timeout is set to the global default (10s).
+
+# Alertmanager configuration
+alerting:
+  alertmanagers:
+  - static_configs:
+    - targets:
+      # - alertmanager:9093
```
Comment (justinjc) [inline src/integration/resources/docker/config/prometheus.yml:14]: Nit: if we don't want these, can we remove and put `null` instead?

File: src/integration/resources/docker/config/prometheus.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
+rule_files:
+  # - "first_rules.yml"
+  # - "second_rules.yml"
```
Comment (justinjc) [inline src/integration/resources/docker/config/prometheus.yml:19]: Same as above

File: src/integration/aggregator/aggregator.go [hunk 3]
```diff
@@ -37,8 +37,9 @@ import (
 
 const (
 	// TestAggregatorDBNodeConfig is the test config for the dbnode.
+	// TODO (amainsd): needs to use host.docker.internal for buildkite
 	TestAggregatorDBNodeConfig = `
-db: {}
+"db": {}
```
Comment (justinjc) [inline src/integration/aggregator/aggregator.go:42]: Curious why this is needed?

File: src/integration/resources/inprocess/cluster.go [hunk 4]
```diff
@@ -390,7 +457,7 @@ func GenerateDBNodeConfigsForCluster(
 	)
 	for i := 0; i < int(numNodes); i++ {
 		var cfg dbcfg.Configuration
```
Comment (justinjc) [inline src/integration/resources/inprocess/cluster.go:459]: Nit: probably don't need this line anymore

File: src/integration/resources/inprocess/dbnode_test.go [hunk 5]
```diff
@@ -206,3 +211,23 @@ const defaultDBNodeConfig = `
...
+     - "endpoints": ["http://127.0.0.1:2379"]
+       "zone": "embedded"
+     "service": "m3db"
+     "zone": "embedded"
+     "env": "default_env"
+`
+	var dbCfg dbcfg.Configuration
+	require.NoError(t, yaml.Unmarshal([]byte(yamlCfg), &dbCfg))
+	spew.Dump(dbCfg)
+	//fmt.Printf("%+v\n", dbCfg)
```
Comment (justinjc) [inline src/integration/resources/inprocess/dbnode_test.go:232]: Did you want to keep these debugs?

File: src/integration/resources/docker/config/prometheus.yml (hunk 1 above)
Comment (andrewmains12) [reply src/integration/resources/docker/config/prometheus.yml:14]: Oh uh, this file actually should be a `git mv` (i.e. I didn't really touch it), apologies. Let me clear that up.

File: src/integration/aggregator/aggregator.go (hunk 3 above)
Comment (andrewmains12) [reply src/integration/aggregator/aggregator.go:42]: Ah yeah it's not--cruft, sorry. Fixing.

File: src/integration/resources/inprocess/dbnode_test.go (hunk 5 above)
Comment (andrewmains12) [reply src/integration/resources/inprocess/dbnode_test.go:232]: Gah sorry, I should've taken a pass on these. Cleaning.

File: src/integration/resources/inprocess/cluster.go [hunk 6]
```diff
@@ -122,17 +125,51 @@ func NewClusterConfigsFromConfigFile(
 // NewClusterConfigsFromYAML creates a new ClusterConfigs object from YAML strings
 // representing component configs.
 func NewClusterConfigsFromYAML(dbnodeYaml string, coordYaml string, aggYaml string) (ClusterConfigs, error) {
-	var dbCfg dbcfg.Configuration
+	// "db":
```
Comment (justinjc) [inline src/integration/resources/inprocess/cluster.go:128]: Is this comment block still relevant?

**Review by justinjc — APPROVED** [review]

**Pull request merged**